### PR TITLE
Simplified error handling using Python 3 syntax

### PR DESCRIPTION
**Pull request by Avasam**
https://github.com/pyasn1/pyasn1/commit/81ce8d0d5f1c6ae4662cdc1fe141b7ea0cd28e00 removed support for Python 2, so error handling can be migrated to Python 3 syntax.

~~I've left in `sys.exec_info` that would be removed by https://github.com/pyasn1/pyasn1/pull/62 (modifying those would be invalid Python 2 syntax anyway)~~ (that PR has now been merged)

### CODE REVIEW

File: tests/codec/ber/test_decoder.py [hunk 1]
```diff
@@ -1746,9 +1746,7 @@ def testErrorCondition(self):
             asn1Object = next(decode(stream))
 
         except error.PyAsn1Error:
-            exc = sys.exc_info()[1]
-            assert isinstance(exc, error.PyAsn1Error), (
-                'Unexpected exception raised %r' % (exc,))
+            pass
```
Comment (droideck) [inline tests/codec/ber/test_decoder.py:1749]: What's the reason behind this change? Why do we need to `pass`?

Can we just make it Python3ish and leave as it is?

```python
except error.PyAsn1Error as exc:
    assert isinstance(exc, error.PyAsn1Error), (
        f'Unexpected exception raised {exc!r}')
```

**Review by droideck — CHANGES_REQUESTED** [review]
Besides the minor concern, looks good!

File: tests/codec/ber/test_decoder.py (hunk 1 above)
Comment (Avasam) [reply tests/codec/ber/test_decoder.py:1749]: AFAICT it's dead code. `exc` should always be an instance of `error.PyAsn1Error` given that's the type you're catching.

Comment (droideck) [reply tests/codec/ber/test_decoder.py:1749]: Right, `isinstance` fully passes for any inherited types. Nevermind. :)

**Review by droideck — APPROVED** [review]
